pkg/hostagent/services.go

```diff
@@ -326,8 +350,41 @@ func (agent *HostAgent) updateServiceDesc(external bool, as *v1.Service,
 	if hasValidMapping {
 		if (ok && !reflect.DeepEqual(existing, ofas)) || !ok {
 			agent.opflexServices[ofas.Uuid] = ofas
-			return true
+			if agent.config.AciVmmDomainType == "OpenShift" {
+				if !external {
+					for _, v := range agent.ocServices {
+						if v.Name != as.ObjectMeta.Name &&
+							v.Namespace != as.ObjectMeta.Namespace {
+							continue
+						}
+						var ingressIp string
```

Comment:
This processing is specific to just the ingressIp. We need the ability to be able to add other IPs as well. The plan was to have a list, each element of which would be a 3-tuple: (namespace, service-name, ip-addres)

That way we can populate the list with all the known the desired 3-tuples and generically process them here. Now I think you will need to special processing for every service we add.